lib/discovery/index.js

```diff
@@ -233,64 +235,84 @@ export class Discovery extends TypedEmitter {
    * @param {import('net').Socket} socket
    * @returns {Promise<void>}
    */
+  // Don't think this needs to be async?
   async _onMdnsConnection(socket) {
     if (socket.destroyed) {
-      throw new Error('Socket destroyed')
+      console.error('Socket destroyed')
+      return
     }
 
-    const socketAddress = /** @type {import('net').AddressInfo} */ (
-      socket.address()
-    )
+//     const socketAddress = /** @type {import('net').AddressInfo} */ (
+//       socket.address()
+//     )
 
-    if (!socketAddress.address) {
-      throw new Error('Socket not connected')
+    if (!socket.remoteAddress) {
+      console.error('Socket not connected')
+      return
     }
+    if (!isIpPrivate(socket.remoteAddress)) socket.destroy()
```

Comment:
@tomasciccola this should return here. Also there are no error handlers on socket at this point so this will crash with an unhandled error